### PR TITLE
Implement delete

### PR DESCRIPTION
**Pull request by itdove**
https://github.com/open-cluster-management/library-go/issues/12

I tried this version with the rcm-controller and it works.

### CODE REVIEW

File: cmd/applier/main.go [hunk 1]
```diff
@@ -24,17 +25,23 @@ func main() {
...
+	var timeout int
+	var force bool
 	klog.InitFlags(nil)
 	flag.StringVar(&dir, "d", ".", "The directory containing the templates, default '.'")
 	flag.StringVar(&valuesPath, "values", "values.yaml", "The directory containing the templates, default 'values.yaml'")
 	flag.StringVar(&kubeconfigPath, "k", "", "The kubeconfig file")
 	flag.BoolVar(&dryRun, "dry-run", false, "if set only the rendered yaml will be shown, default false")
 	flag.StringVar(&prefix, "p", "", "The prefix to add to each value names, for example 'Values'")
+	flag.BoolVar(&delete, "delete", false, "if set only the resource defined in the yamls will be deleted, default false")
+	flag.IntVar(&timeout, "t", 5, "Timeout in second to apply one resource, default 1 sec")
```
Comment (hanqiuzh) [inline cmd/applier/main.go:38]: default 5 second?

Comment (itdove) [reply cmd/applier/main.go:38]: 5 sec per transaction (CRUD). You think too much or too little

File: pkg/applier/applier.go [hunk 2]
```diff
@@ -500,6 +644,87 @@ func (a *Applier) Update(
...
+	})
+	if err != nil && !errors.IsNotFound(err) {
+		klog.V(2).Info("Unable to delete:", "Error", err,
+			" Kind: ", u.GetKind(),
+			" Name: ", u.GetName(),
+			" Namespace: ", u.GetNamespace())
+		return err
+	}
+	if a.applierOptions.ForceDelete &&
+		u.GetKind() != reflect.TypeOf(apiextensions.CustomResourceDefinition{}).Name() {
```
Comment (hanqiuzh) [inline pkg/applier/applier.go:688]: so not force deleting crd, that make sense, and how about namespace?

Comment (hanqiuzh) [reply pkg/applier/applier.go:688]: and it would be nice to doc in either the flags' help message?

Comment (itdove) [reply pkg/applier/applier.go:688]: added in the doc

**Pull request merged**